components/clarinet-files/src/devnet_diff.rs

```diff
@@ -0,0 +1,260 @@
+                    orders
+                        .into_iter()
+                        .map(|pso| {
+                            format!(
+                                "{}-{}-{}-{}-{}",
+                                pso.start_at_cycle,
+                                pso.duration,
+                                pso.wallet,
+                                pso.slots,
+                                pso.btc_address
+                            )
+                        })
+                        .collect::<Vec<_>>()
+                        .join(",")
+                }),
+            },
+        ]
+    }
+
+    /// Check if any significant fields are different
+    pub fn is_compatible(
+        &self,
+        user_config: &DevnetConfig,
+    ) -> Result<bool, Vec<(String, String, String)>> {
+        let default_config = DevnetConfig::default();
+        let mut errors = Vec::new();
+        for field in &self.significant_fields {
+            let default_value = (field.extractor)(&default_config);
+            let user_value = (field.extractor)(user_config);
+
+            if default_value != user_value {
+                if field.name == "pox_stacking_orders" {
+                    let stacking_errors = self.clean_stacking_orders(user_value, default_value);
+                    for (field_name, user_val, default_val) in stacking_errors {
+                        errors.push((field_name, user_val, default_val));
+                    }
+                } else {
+                    errors.push((field.name.clone(), user_value, default_value))
+                }
+            }
+        }
+        if errors.is_empty() {
+            Ok(true)
+        } else {
+            Err(errors)
+        }
+    }
+
+    fn clean_stacking_orders(
+        &self,
+        user_value: String,
+        default_value: String,
+    ) -> Vec<(String, String, String)> {
+        let mut result = Vec::new();
+        let user_orders: Vec<String> = user_value.split(",").map(String::from).collect();
+        let default_orders: Vec<String> = default_value.split(",").map(String::from).collect();
+        let constructed_user_orders = construct_stacking_orders(&user_orders);
+        let constructed_default_orders = construct_stacking_orders(&default_orders);
+        for (btc_address, user_order) in constructed_user_orders {
+            let default_order = constructed_default_orders.get(&btc_address).unwrap();
+            for (field, user_val) in user_order {
+                let default_val = default_order
+                    .iter()
+                    .find(|(f, _)| *f == field)
+                    .map(|(_, v)| v)
+                    .unwrap();
+                if default_val != &user_val {
+                    result.push((field, user_val, default_val.clone()));
+                }
+            }
+        }
+        result
+    }
+}
+
+fn construct_stacking_orders(orders: &Vec<String>) -> HashMap<String, Vec<(String, String)>> {
```

Comment:
As a general rule, it's better to pass slices if you aren't going to consume/modify a `Vec`:

```suggestion
fn construct_stacking_orders(orders: &[String]) -> HashMap<String, Vec<(String, String)>> {
```